mantidproject/mantid
# Fix issue with Elemental analysis not supporting legacy data

The elemental analysis GUI had difficulty dealing with legacy data. For instance, if the data set was missing the delayed response, an error was raised, which prevented any data from being plotted. This issue was encountered by Adrian Hillier while processing some old data. 

This PR fixes this issue, by associating a workspace (of Y(x)=0) to the missing data. This enables  elemental analysis to continue as expected.

**Report to:** Adrian Hillier 


**To test:**
Make of a copy of some elemental analysis data (this is not readily available)
Delete the delayed response (files ending in 20)
Open up the elemental analysis GUI
Add the modified run data to the user directories
Load the modified run (the one where you've deleted the delayed response)
The GUI should be able to handle this. Check that all the plots work. The delayed response should be plotted with (Y=0)



Fixes #27430.




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
